library/core/src/mem/maybe_uninit.rs

```diff
@@ -573,6 +574,27 @@ impl<T> MaybeUninit<T> {
         }
     }
 
+    /// Drops the contained value in place.
+    ///
+    /// If you have ownership of the `MaybeUninit`, you can use [`assume_init`] instead.
+    ///
+    /// # Safety
+    ///
+    /// Calling this when the content is not yet fully initialized causes undefined
+    /// behavior: it is up to the caller to guarantee that the `MaybeUninit<T>` really
+    /// is in an initialized state.
+    ///
+    /// This function runs the destructor of the contained value in place.
+    /// Afterwards, the memory is considered uninitialized again, but remains unmodified.
+    ///
+    /// [`assume_init`]: MaybeUninit::assume_init
+    #[unstable(feature = "maybe_uninit_extra", issue = "63567")]
+    pub unsafe fn drop(&mut self) {
```

Comment:
Given that we otherwise have `assume_init` in all methods that require things to be initialized, maybe it would be a good idea to call this `drop_assume_init` or `assume_init_drop` or so?

Reply:
Yes, maybe. Since the same question is open for `read` (https://github.com/rust-lang/rust/issues/63567#issuecomment-521357943), I figured I'd use just `drop` for now. Then they can be changed together later if a different name is preferred.

Reply:
`read` was added before `assume_init` had that name. I don't feel like it is a good idea to add more methods with "legacy" names.

Reply:
Then it's probably best to first finish the decision on the new name for `read` before merging this.

Reply:
Do you want to just rename `read` in this PR?

Reply:
Ah, I was under the impression there was still more discussion required for the new name. But looking at https://github.com/rust-lang/rust/pull/76047 it looks like `assume_init_*` was already decided upon.

Sure, will update the names. :)